cfgmgmtcamp2025-Containerizing_Foreman_deployments_take_42.md

```diff
@@ -0,0 +1,396 @@
+
+---
+
+## Foreman container
+
+```[|7]
+# cat /etc/containers/systemd/foreman.container
+[Container]
+ContainerName=foreman
+HostName=quadlet.example.com
+Image=quay.io/evgeni/foreman-rpm:nightly
+Network=host
+Secret=foreman-database-url,type=env,target=DATABASE_URL
+Secret=foreman-settings-yaml,type=mount,
+       target=/etc/foreman/settings.yaml
+…
+Secret=foreman-client-cert,type=mount,
+       target=/etc/foreman/client_cert.pem
+…
+```
+
+Note:
+* Conceptually, this looks exactly like the Candlepin container
+* One noteworthy new thing is `type=env`
+* And yes, this is missing passing in the plugins variable
+
+---
+
+## Dynflow containers
+
+```[|1,3|10|6-8]
+# cat /etc/containers/systemd/dynflow-sidekiq@.container
+[Container]
+ContainerName=dynflow-sidekiq-%i
+Environment=DYNFLOW_SIDEKIQ_SCRIPT=
+  /usr/share/foreman/extras/dynflow-sidekiq.rb
+Exec=/usr/libexec/foreman/sidekiq-selinux -e production \
+     -r /usr/share/foreman/extras/dynflow-sidekiq.rb \
+     -C /etc/foreman/dynflow/%i.yml
+HostName=quadlet.example.com
+Image=quay.io/evgeni/foreman-rpm:nightly
+Network=host
+Secret=…
+```
+
+Note:
+* We can do template containers! Just like systemd
+* This can be instantiated as `@orchestrator`, `@worker` etc
+* It's actually using the very same container image as the Foreman container
+* as it's mainly the same code running, just a different entry point
+
+---
+
+## Which containers?
+
+<div class="mermaid">
+  <pre>
+classDiagram
+    Foreman ..> Foreman_Proxy
+    Foreman ..> Candlepin
+    Foreman ..> Pulp
+    Foreman ..> Redis
+    Pulp ..> Redis
+    httpd ..> Foreman
+    httpd ..> Pulp
+    Pulp ..> PostgreSQL
+    Foreman ..> PostgreSQL
+    Candlepin ..> PostgreSQL
+    dynflow_orchestrator ..> PostgreSQL
+    dynflow_orchestrator ..> Redis
+    dynflow_worker ..> PostgreSQL
+    dynflow_worker ..> Redis
+    dynflow_worker ..> Pulp
+    dynflow_worker ..> Candlepin
+
+    style Foreman fill:#f96
+    style Foreman_Proxy fill:#096
+    style Pulp fill:#f96
+    style Candlepin fill:#f96
+    style dynflow_orchestrator fill:#f96
+    style dynflow_worker fill:#f96
+  </pre>
+</div>
+
+Note:
+* Pulp is replaced by containers offered by the Pulp project
+* Foreman Proxy is only here for Katello/Pulp communication, not a full proxy
+* The containers run as systemd services, so start/stop/journal still works
+
+---
+
+## are we there yet?
+
+<span class="fragment">No.</span>
+<span class="fragment">But we know which question we need to answer.</span>
+
+---
+
+## installation / configuration
+
+* our research used Ansible
+* our current Installer is Puppet
+* is this a chance to simplify?
+
+Note:
+* "everything is a container" could make things easier to deploy
+* can we configure more things using environment variables?
+* but in reality we're not there yet
+
+---
+
+## migrations / upgrades
+
+* existing setups will require migration
+* the installer decision will influence this
+
+---
+
+## PostgreSQL
```

Comment:
Like PostgreSQL, httpd is also an open research point. Especially when we look at external auth.